flow-report/hooks.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+// TODO(FR-COMPAT): Use context router to handle steps.
+export function useCurrentStep() {
```

Comment:
want to pass the flow result here to resolve the selected report too? that way we can check for inbounds of array, reduce consumer burden, etc

`return {currentLhrIndex, currentLhr}` or similar

Reply:
I'm tempted to stick the `flowResult` in a context so we don't have mine it through every component every time we use this hook. WDYT?

Reply:
`flowResult` in context sounds good

my overall comment stands on the `{currentLhrIndex, currentLhr}` bit just replace "pass the flow result" with "use `useContext(FlowResult)`" :)